teps/0073-simplify-metrics.md

```diff
@@ -0,0 +1,427 @@
+time series because we use `pipelinerun` and `taskrun` labels/tags. This causes
+unbounded cardinality which isn't recommended for systems like prometheus.
+While we can expect `Pipeline` or `Task` object to remain fairly constant,
+`pipelinerun` or `taskrun` object will continue to increase which lead to
+cardinality explosion. In systems, where `Pipeline` and `Task` are expected to
+grow, labels based on them can also cause cardinality explosion. 
+
+### Large metrics volume
+
+Currently, there is too much metrics per
+`TaskRun`/`PipelineRun`. According to
+[metrics.md](https://github.com/tektoncd/pipeline/blob/master/docs/metrics.md),
+the metrics count of a `PipelineRun` with `n` `TaskRun` is
+approximately `15*(n+1)+n`. Majority of metrics comes from histogram.
+The amount of metrics is huge while cluster is under stress. This
+amount of metrics will cause severe cluster performance degration if
+users have monitoring system running unthrottled.
+
+Large metrics volume may also cause metrics loss. Some monitoring
+system has limitation on metrics count.  For example, the maximum
+number of Prometheus metrics that the agent can consume from the
+target is 10000 according to [Prometheus Metrics
+Limitation](https://docs.sysdig.com/en/limit-prometheus-metric-collection.html).
+Once metrics count reaches limitation, new metrics will be dropped and
+we cannot receive them anymore.
+
+### Fine granularity
+
+Currently, Tekton pipelines collect metrics at `TaskRun` and
+`PipelineRun` level. Users usually find it too noisy and want
+high-level and aggregated metrics.
+
+### Goals
+
+- Reduce the metrics volume
+- Allow users to configure the metrics granularity as they want
+- Simplify metrics so that users can see what they care about easily
+
+### Non-Goals
+
+- Change the behavior of existing metrics ingestor (Prometheus, Stackdriver etc)
+- Add support to a new metrics ingestor to handle a large amount of metrics
+- Build a new metrics ingestor to handle a large amount of metrics
+
+### Use Cases (optional)
+
+Cluster admin users are able to configure Tekton pipelines to produce
+high level metrics and focus on overall status to make sure Tekton
+works fine.  While single users who care about statistics of
+individual `TaskRun` can still keep the fine-grained metrics.
+
+## Requirements
+
+<!--
+Describe constraints on the solution that must be met. Examples might include
+performance characteristics that must be met, specific edge cases that must
+be handled, or user scenarios that will be affected and must be accomodated.
+-->
+
+## Proposal
+
+Coarse-grained metrics should be provided to satisfy the users'
+needs. This can be implemented in following ways: 
+- changing `PipelineRun` or `TaskRun` level metrics to `Task` or `Pipeline`
+  level
+- changing `PipelineRun` or `TaskRun` level metrics to namespace level
+- changing metrics type from histogram to gauge in case of `TaskRun` or
```

Comment:
I think that the duration metric could even be removed here. You most likely only see a task run once so it is not very meaningful to have an average duration over only one element.